### ISSUE TITLE
Naseptavac u pouze volne dostupnych dokumentu

### ISSUE THREAD

**Issue by honza-rychtar** [moravianlibrary/kramerius-for-ios#114]
Pokud si v nastaveni zatrhnu *zobrazovat jen volne dostupne*, tak mi naseptavac nabizi i neverejne dokumenty, ktere pak po odpaleni nenajde.
Pokud by to bylo na urovni filtru, tak ok. Ale tohle je prepinac v nastaveni - tim by se melo zmenit chovani cele aplikace.

**Comment by ovyhlidal** [comment]
Meni se tim chovani cele aplikace, jen jsem v naseptavaci toto nastaveni ignoroval. Opraveno bude v dnesnim releasu. Jeste bych se chtel domluvit na chovani v tomto pripade:
1. V nastaveni si zapnu abych videl i nedostupne dokumenty
2. Vyhledam si nedostupny dokument
3. Aplikace jej nedovoli otevrit
4. V nastaveni zmenim nastaveni viditelnosti dokumentu
5. Vyhledam stejny dokument, naseptavac mi ze serveru nic nenasepta, ale z lokalni historie dostanu platny vyskyt stringu. Kliknu na nej, provede se plnohodnotne vyhledavani, ale tentokrat nic nenajdu (prepnul jsem si viditelnost dokumentu) a aplikace mi napise, ze nic nenasla. 
@honza-rychtar Chova se tvuj klient stejne?

**Comment by zabak** [comment]
@honza-rychtar mohl bys odpovědět? Ve starší verzi aplikace pro Android toto nejde reprodukovat, protože funguje trochu jinak. Osobně si myslím, že výše pospané chování je korektní.

**Comment by honza-rychtar** [comment]
K bodu 3)
Nekde jsme to resili, ale nechapu, proc se neda neverejny dokument rozkliknout. 
Bud neverejne dokumenty z aplikace vyhodit uplne, nebo mit moznost je rozkliknout a varovani zobrazit az potom., Kazdopadne nedelat varovani na urovni policy priznaku, ale 401 odpovedi pri vyzadani obrazku.

Duvodu je vic.
1) Pokud si nemuzu dokument otevrit, tak si nemam jak zobrazit metadata.
2) Priznak policy (bohuzel) nijak neovlivnuje konecnou dostupnost dokumentu. Takze dokument muze mit policy priznak private, ale dokument (na urovni obrazovych dat a ocr) je verejne dostupny. Podivej se treba na Narodni muzeum, tam je vse podle policy neverejne, ale vetsina dokumentu otevrit jde a je to tak spravne, protoze uz nepodlehaji autorskemu zakonu. 
3) Kramerius umoznuje zpristupneni dokumentu pres tzv. pohyblivou zed, kdy se zpristupni dokumenty starsi nez X. Tohle ovlivni pritup obrazovych dat, ale neovlivni policy priznaky.
Je to nesmysl, ale proste to tak funguje. Mozna uz se to zmenilo, ale starsi dokumenty zpristupnene pres pohyblivou zed budou stale v nekonzistentnim stavu.
4) Kramrius umoznuje nastavit pristup k neverejnym dokumentum pres IP adresu. Takze na vybranych pocitacich knihoven jsou v klientech dokumenty techto knihoven dostupne, i kdyz maji policy private.
MZK to treba vyuzuva i pro mobilni aplikace, kdy jsou tablety s Androidem prihlaseny k wifi s IP adresou, ktera ma v Krameriovi povoleny pristup k neverejnym dokumentum.

Abych to shrnul, tak policy priznak by mel slouzit pouze k fitrovani a zobrazeni zamecku, ale ne k rizeni pristupu v aplikaci, stejne jako jej k rizeni pristupu nepouziva Kramerius.

Jinak k celkovemu navrzenemu chovani - ano, pokud se bude "pouze verejne" resit pres nasataveni.
Jakmile bude mozne prepinat mezi pouze verejne/vse rovnou u vysledku hledani, tak potom nechat i ve stavu "pouze verejne" naseptat  neverejne dokumenty.

**Comment by zabak** [comment]
Souhlasím, neveřejný dokument musí jít rozkliknout, přístupnost se řeší na úrovni konkrétních stránek.

**Comment by ovyhlidal** [comment]
Ok, tak to opet predelam. Puvodne to tak bylo, ze se zobrazovala hlaska az po rozkliknuti. Nejsem si jist, proc se to predelalo.

**Comment by zabak** [comment]
Upřesním. Na androidu je to tak, že u karet s jednotlivými dokumenty je v pravém horním rohu malé menu, kde se dá rozkliknout položky Podrobnosti, Otevřít a Sdílet. Tj. na androidu neuvidím thumbnaily jednotlivých stránek (a kdyby tam byla nějaká volná, asi se na ni nijak nedostanu), ve webové verzi ano. Stačilo by tedy přidat na kartu dokumentu buď obdobné minimenu, nebo zatím jen i v kroužku vedoucí na stránku s informacemi o dokumentu.
